runtime/src/main/java/org/corfudb/runtime/object/StreamViewSMRAdapter.java

```diff
@@ -180,8 +182,8 @@ public Stream<SMREntry> streamUpTo(long maxGlobal) {
     @Override
     public Stream<SingleAddressUpdates> streamUpToInList(long maxGlobal) {
         return streamView.streamUpTo(maxGlobal)
-                .filter(m -> m.getType() == DataType.DATA)
-                .filter(m -> m.getPayload(runtime) instanceof ISMRConsumable
+                .filter(m -> m.getType() == DataType.HOLE || m.getType() == DataType.DATA)
+                .filter(m -> m.getType() == DataType.HOLE || m.getPayload(runtime) instanceof ISMRConsumable
                         || m.hasCheckpointMetadata())
```

Comment:
This seems a little superfluous: unlike `streamUpTo` , `streamUpToInList` calls `dataCheckpointMapper` which can handle holes, so why apply the filter in the first place?

In other words, `streamUpTo` must return a `SMREntry` so it must filter holes, but `streamUpToInList` returns a `SingleAddressUpdates` which represents a hole as a empty collection.

Reply:
Good point. Fixed.